book/pixi-exercise.md

```diff
@@ -0,0 +1,182 @@
+# Pixi: exercises
+This page contains exercises to help you practice the concepts learned in the Pixi introduction. Each exercise is designed to reinforce your understanding of Pixi's features and capabilities.
+
+## Exercise 1: Creating a project
+1. Create a new Pixi project named `my-project` using the command line.
+2. Add `python` as a dependency to your project.
+3. Use the Pixi installed Python to print `"Hello, Pixi!"` to the console.
+4. Run a command to print the version of Python used in your project. You can see the location of the interpreter by running: `import sys;print(sys.executable)`
+
+:::{hint} Solution
+:class: dropdown
+```bash
+# 1
+pixi init my-project
+cd my-project
+# 2
+pixi add python
+# 3
+pixi run python -c 'print("Hello, Pixi!")'
+# 4
+pixi run python -c 'import sys; print(sys.version); print(sys.executable)'
+```
+:::
+
+## Exercise 2: Dependencies
+1. Make your project work for Windows, macOS, and Linux.
+2. Add `scipy` as a dependency through the command line.
+3. Add `numpy` as a dependency through the command line.
+4. Add `fastqc` as a dependency from `bioconda`. Hint: the channel has to be added first.
+5. Add `pandas` as a `pypi` dependency through the command line.
+6. Add a `pypi` + `git` dependency on [`pytest`](https://github.com/pytest-dev/pytest) to your project.
+7. Visualize the locked dependencies in the command line.
+
+:::{hint} Solution
+:class: dropdown
+```bash
+# 1
+pixi workspace platform add win-64 osx-arm64 osx-64 linux-64
+# 2
+pixi add scipy
+# 3
+pixi add numpy
+# 4
+pixi workspace channel add bioconda
+pixi add bioconda::fastqc
+# 5
+pixi add pandas --pypi
+# 6
+pixi add pytest --pypi --git https://github.com/pytest-dev/pytest
```

Comment:
Though you already give the full command line structure to create the workspace in each exercise, what do you think about providing the final `pixi.toml` that would be generated from these commands in the the `dropdown` hidden Solution as well? I could imagine this might be helpful for people that edited things by hand in the `pixi.toml` and either have a typo or want to understand what they put in the wrong place?

Reply:
That would be easy!